### PR TITLE
Use tags

### PR DESCRIPTION
**Pull request by dantleech**
Deprecate the existing `context` option and replace it with `tag`.

```bash
$ phpbench run --tag=impl_a --store
$ phpbench run --tag=impl_b --store
$ phpbench run --tag=impl_c --store
$ phpbench report --uuid=tag:impl_a --uuid=tag:impl_b --uuid=tag:impl_c --report='{extends: compare, compare: tag, break:[]}'

+--------------+----------+--------+--------+------+-----------------+-----------------+-----------------+
| benchmark    | subject  | groups | params | revs | tag:impl_a:mean | tag:impl_b:mean | tag:impl_c:mean |
+--------------+----------+--------+--------+------+-----------------+-----------------+-----------------+
| HashingBench | benchMd5 |        | []     | 1000 | 0.957μs         | 0.939μs         | 0.952μs         |
+--------------+----------+--------+--------+------+-----------------+-----------------+-----------------+
```

### CODE REVIEW

File: extensions/dbal/lib/Storage/Driver/Dbal/Persister.php [hunk 1]
```diff
@@ -36,7 +36,8 @@ public function persist(SuiteCollection $collection)
             $summary = $suite->getSummary();
             $id = $this->insertUpdate($conn, 'run', [
                 'uuid' => $suite->getUuid(),
-                'context' => $suite->getContextName(),
+                'tag' => $suite->getTag(),
+                'tag' => $suite->getTag(),
```
Comment (dantleech) [inline extensions/dbal/lib/Storage/Driver/Dbal/Persister.php]: only once

File: extensions/dbal/lib/Storage/Driver/Dbal/Schema.php [hunk 2]
```diff
@@ -40,7 +40,7 @@ private function createRun()
         $table = $this->createTable('run');
         $table->addColumn('id', 'integer', ['autoincrement' => true]);
         $table->addColumn('uuid', 'string');
-        $table->addColumn('context', 'string', ['notnull' => false]);
+        $table->addColumn('tag', 'string', ['notnull' => false]);
```
Comment (dantleech) [inline extensions/dbal/lib/Storage/Driver/Dbal/Schema.php]: probably going to remove the dbal storage at some point, worth having the BC break? (only needs a schema update ..)

File: docs/benchmark-runner.rst [hunk 3]
```diff
@@ -201,16 +201,16 @@ You can compare the results of two or more sets of results using the `compare`
...
 
     $ # .. configure for implementation A
-    $ phpbench run --context="Impl. A" --dump-file=impl-a.xml
+    $ phpbench run --tag="Impl. A" --dump-file=impl-a.xml
     $ # .. configure for implementation B
-    $ phpbench run --context="Impl. B" --dump-file=impl-b.xml
+    $ phpbench run --tag="Impl. B" --dump-file=impl-b.xml
     $ # .. configure for implementation C
-    $ phpbench run --context="Impl. C" --dump-file=impl-c.xml
+    $ phpbench run --tag="Impl. C" --dump-file=impl-c.xml
```
Comment (dantleech) [inline docs/benchmark-runner.rst]: invalid syntax!

**Comment by dantleech** [comment]
@fhein this feature should make it slightly easier to compare benchmarks.

**Comment by fhein** [comment]
Yes, it does! Thank you very much! :+1:

**Pull request merged**